tst/providers/MeetingProvider/MeetingManager.test.ts

```diff
@@ -0,0 +1,120 @@
+    mockMeetingManager.setupAudioVideoObservers = jest.fn().mockReturnValue({});
+    mockMeetingManager.setupDeviceLabelTrigger = jest.fn().mockReturnValue({});
+    mockMeetingManager.listAndSelectDevices = jest.fn().mockReturnValue({});
+    mockMeetingManager.publishAudioVideo = jest.fn().mockReturnValue({});
+    mockMeetingManager.setupActiveSpeakerDetection = jest
+      .fn()
+      .mockReturnValue({});
+    // @ts-ignore
+    mockMeetingManager.publishMeetingStatus = jest.fn().mockReturnValue({});
+  });
+
+  describe('join', () => {
+    it('should call DefaultMeetingSession', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(DefaultMeetingSession).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call createLogger if MeetingManagerConfig does not pass one', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(mockMeetingManager.createLogger).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call removeObserver on AudioVideoFacade', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(
+        mockMeetingManager?.audioVideo?.removeObserver
+      ).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call addObserver on AudioVideoFacade', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(mockMeetingManager?.audioVideo?.addObserver).toHaveBeenCalledTimes(
+        1
+      );
+    });
+
+    it('should call setupAudioVideoObservers', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(mockMeetingManager.setupAudioVideoObservers).toHaveBeenCalledTimes(
+        1
+      );
+    });
+
+    it('should call setupDeviceLabelTrigger', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(mockMeetingManager.setupDeviceLabelTrigger).toHaveBeenCalledTimes(
+        1
+      );
+      });
+
+    it('should call listAndSelectDevices', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(mockMeetingManager.listAndSelectDevices).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call publishAudioVideo', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(mockMeetingManager.publishAudioVideo).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call setupActiveSpeakerDetection', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      expect(
+        mockMeetingManager.setupActiveSpeakerDetection
+      ).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call publishMeetingStatus', async () => {
+      await mockMeetingManager.join(mockMeetingJoinData);
+      // @ts-ignore
+      expect(mockMeetingManager.publishMeetingStatus).toHaveBeenCalledTimes(1);
+    });
+
+    it('should use the MeetingSessionConfiguration provided via params', async () => {
+      mockMeetingManager.meetingSessionConfiguration = null;
+
```

Comment:
nit: Drop these extra newlines :) (Before and after `await`)

Reply:
Done